packages/core/src/auth/credentials/sharedCredentialsFile.ts

```diff
@@ -8,15 +8,24 @@
 import { join } from 'path'
 import { EnvironmentVariables } from '../../shared/environmentVariables'
 import { SystemUtilities } from '../../shared/systemUtilities'
+import { isValidPath } from '../../shared/utilities/pathUtils'
 
 export function getCredentialsFilename(): string {
     const env = process.env as EnvironmentVariables
 
-    return env.AWS_SHARED_CREDENTIALS_FILE || join(SystemUtilities.getHomeDirectory(), '.aws', 'credentials')
+    if (env.AWS_SHARED_CREDENTIALS_FILE && isValidPath(env.AWS_SHARED_CREDENTIALS_FILE)) {
+        return env.AWS_SHARED_CREDENTIALS_FILE
+    }
+
+    return join(SystemUtilities.getHomeDirectory(), '.aws', 'credentials')
```

Comment:
join() is never needed for literals, and reduces readability. (same comment for all other cases in this PR, and in the future)

```suggestion
    return join(SystemUtilities.getHomeDirectory(), '.aws/credentials')
```

Reply:
NIce find :). Will update in the next revision.